core/src/main/java/com/linecorp/armeria/server/RouteBuilder.java

```diff
@@ -0,0 +1,351 @@
+        this.methods = ImmutableSet.copyOf(Sets.immutableEnumSet(requireNonNull(methods, "methods")));
+        return this;
+    }
+
+    /**
+     * Sets {@link MediaType}s that this {@link Route} consumes. If not set, the the mapped {@link Service}
+     * accepts {@link HttpRequest}s that have any {@link HttpHeaderNames#CONTENT_TYPE}.
+     * In order to get this work, {@link #methods(Iterable)} should be set.
+     */
+    public RouteBuilder consumes(MediaType... consumeTypes) {
+        consumes(ImmutableList.copyOf(requireNonNull(consumeTypes, "consumeTypes")));
+        return this;
+    }
+
+    /**
+     * Sets {@link MediaType}s that this {@link Route} consumes. If not set, the mapped {@link Service} accepts
+     * {@link HttpRequest}s that have any {@link HttpHeaderNames#CONTENT_TYPE}.
+     * In order to get this work, {@link #methods(Iterable)} should be set.
+     */
+    public RouteBuilder consumes(Iterable<MediaType> consumeTypes) {
+        ensureUniqueMediaTypes(consumeTypes, "consumeTypes");
+        consumes = ImmutableList.copyOf(consumeTypes);
+        return this;
+    }
+
+    /**
+     * Sets {@link MediaType}s that this {@link Route} produces. If not set, the mapped {@link Service} accepts
+     * {@link HttpRequest}s that have any {@link HttpHeaderNames#ACCEPT}.
+     * In order to get this work, {@link #methods(Iterable)} should be set.
+     */
+    public RouteBuilder produces(MediaType... produceTypes) {
+        produces(ImmutableList.copyOf(requireNonNull(produceTypes, "produceTypes")));
+        return this;
+    }
+
+    /**
+     * Sets {@link MediaType}s that this {@link Route} produces. If not set, the mapped {@link Service} accepts
+     * {@link HttpRequest}s that have any {@link HttpHeaderNames#ACCEPT}.
+     * In order to get this work, {@link #methods(Iterable)} should be set.
+     */
+    public RouteBuilder produces(Iterable<MediaType> produceTypes) {
+        ensureUniqueMediaTypes(produceTypes, "produceTypes");
+        produces = ImmutableList.copyOf(produceTypes);
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link Route} based on the properties of this builder.
+     */
+    public Route build() {
+        checkState(pathMapping != null, "Should set a path before calling this.");
+        if ((!consumes.isEmpty() || !produces.isEmpty()) && methods.isEmpty()) {
+            throw new IllegalStateException("Should set methods if consumes or produces is not empty." +
+                                            " consumes: " + consumes + ", produces: " + produces);
+        }
+        return new DefaultRoute(pathMapping, methods, consumes, produces);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(pathMapping, methods, consumes, produces);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
Think we always do `if (!(o instanceof RouteBuilder))`

Reply:
Copied from another class without much thinking. Let me change all those classes to reduce the same mistake. 😉